libs/common/src/errors/document-not-updated.error.ts

```diff
@@ -0,0 +1,14 @@
+import { DatabaseError } from './database.error';
+
+/**
+ * This error is thrown when a document is not successfully updated.
+ * It is not thrown by default by the db-client,
+ */
+export class DocumentNotUpdatedError extends DatabaseError {
+  constructor(documentId?: string) {
+    const message = documentId ? `Document update failed for document with id: ${documentId}` : 'Document update failed';
```

Comment:
```suggestion
    const message = documentId ? `Document update failed for document with id: ${documentId}` : 'Document update failed with undefined document ID';
```

Reply:
I'm not sure we'll ever have an undefined document id -- I was using the ternary to keep the behaviour of this error in the same pattern of other existing database errors

Reply:
Makes sense to be consistent